### PR TITLE
Add "preview_on_sphere" setting of material editor plugin

### LINKED ISSUE #36228: Material preview reset to sphere whenever shader is modified
**Godot version:**
v3.2.stable.official

**OS/device including version:**
Windows 10 Pro

**Issue description:**
A material can be previewed on either a sphere or a cube in the Inspector. Whenever a shader is modified, the material preview is reset to sphere. This results in having to repeatedly change the material preview back to cube when working on a shader.

**Steps to reproduce:**
1. Create a new shader material (either Shader or VisualShader).
2. Change the material preview to a cube.
3. Modify the shader (eg rewire nodes or change a value)

The material preview will be reset to a sphere.

### PR DESCRIPTION
**Pull request by wazzy47**
This PR fixes #36228

### CODE REVIEW

File: editor/plugins/material_editor_plugin.cpp [hunk 1]
```diff
@@ -95,13 +95,17 @@ void MaterialEditor::_button_pressed(Node *p_button) {
 		sphere_instance->hide();
 		box_switch->set_pressed(true);
 		sphere_switch->set_pressed(false);
+		EditorSettings::get_singleton()->set_setting("_preview_on_sphere", false);
+		EditorSettings::get_singleton()->save();
```
Comment (akien-mga) [inline editor/plugins/material_editor_plugin.cpp:99]: This shouldn't be saved as a setting, but as project metadata.
```suggestion
		EditorSettings::get_singleton()->set_project_metadata("inspector_options", "material_preview_on_sphere", false);
```

Comment (wazzy47) [reply editor/plugins/material_editor_plugin.cpp]: changed in 60b14c2

**Comment by wazzy47** [comment]
Guess it was too early for squashing... also added buttons setup for the case when box is on the preview, lines 206-207

**Comment by akien-mga** [comment]
Thanks and congrats for your first merged contribution!

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.2.

**Pull request merged**